lib/mongo/server/connection_pool/queue.rb

```diff
@@ -88,7 +87,7 @@ def disconnect!
         # @since 2.0.0
         def enqueue(connection)
           mutex.synchronize do
-            queue.unshift(connection)
+            queue.unshift(connection.record_checkin!)
             resource.broadcast
           end
         end
```

Comment:
I think all connections in the pool, even the initial ones, must have `last_checkin` set. Otherwise, those connections will never be disconnected according to maxIdleTime until they are checked out and checked back in.